packages/react-native-codegen/src/parsers/flow/modules/index.js

```diff
@@ -68,6 +68,26 @@ function nullGuard<T>(fn: () => T): ?T {
   return fn();
 }
 
+const UnsupportedObjectPropertyTypeToInvalidPropertyValueTypeMap = {
+  FunctionTypeAnnotation: 'FunctionTypeAnnotation',
+  VoidTypeAnnotation: 'void',
+  PromiseTypeAnnotation: 'Promise',
+};
+
+function throwUnsupportedObjectPropertyValueTypeAnnotationParserError(
+  moduleName,
+  propertyValue,
+  propertyKey,
+  type,
+) {
+  throw new UnsupportedObjectPropertyValueTypeAnnotationParserError(
+    moduleName,
+    propertyValue,
+    propertyKey,
+    type,
+  );
+}
+
```

Comment:
This can be moved into and `error-utils.js` file which is shared between typescript and flow. As you can see, the logic is exactly the same.

Reply:
Yes, I was waiting for the other PR to be merged first.